clean.py

```diff
@@ -62,16 +62,28 @@
 assert MAPS_KEYS.isdisjoint(FILTERS_KEYS)
 assert (MAPS_KEYS | FILTERS_KEYS).isdisjoint(DEDUPS_KEYS)
 
-def quick_size_estimation(ds, content_key="text"):
+def get_size_per_example(examples, content_key):
+    size_values = [len(text.encode()) for text in examples[content_key]]
+    examples["bytes_len"] = size_values
+    return examples
+
+def quick_size_estimation(ds, 
+    num_proc: int,
+    batch_size: int,
+    content_key:str ="text"):
     rng = default_rng(1991)
     subset_size = min(10000, len(ds))
     indices = rng.choice(len(ds), size=subset_size, replace=False, shuffle=False)
     partial_ds = ds.select(indices)
     ratio = float(len(ds)) / float(subset_size)
 
-    len_bytes = 0
-    for content in partial_ds[content_key]:
-        len_bytes += len(content.encode())
+    partial_ds = partial_ds.map(
+        partial(get_size_per_example, content_key=content_key),
+        batched=True, 
+        num_proc=num_proc,
+        batch_size=batch_size
```

Comment:
You can remove the columns. Typically this copies all the text in the cache. So you probably only need the `bytes_len` field. We can allways concatenate with original dataset if we need text matching.

Reply:
Good idea, I've added `input_columns` and `remove_columns` to the map method